skxray/tests/test_diff_roi_choice.py

```diff
@@ -0,0 +1,187 @@
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+import numpy as np
+import logging
+logger = logging.getLogger(__name__)
+from numpy.testing import (assert_array_equal, assert_array_almost_equal,
+                           assert_almost_equal)
+import sys
+
+from nose.tools import assert_equal, assert_true, raises
+
+import skxray.diff_roi_choice as diff_roi
+
+from skxray.testing.decorators import known_fail_if
+import numpy.testing as npt
+
+
+def test_roi_rectangles():
+    num_rois = 3
+    detector_size = (15, 26)
+    roi_data = np.array(([2, 2, 6, 3], [6, 7, 8, 5], [8, 18, 5, 10]),
+                        dtype=np.int64)
+
+    roi_inds, num_pixels, pixel_list = diff_roi.roi_rectangles(num_rois,
+                                                               roi_data,
+                                                               detector_size)
+    ty = np.zeros(detector_size).ravel()
+    ty[pixel_list] = roi_inds
+    num_pixels_m = (np.bincount(ty.astype(int)))[1:]
+
+    re_mesh = ty.reshape(*detector_size)
+    for i, (col_coor, row_coor, col_val, row_val) in enumerate(roi_data, 0):
+        ind_co = np.column_stack(np.where(re_mesh == i + 1))
+
+        left, right = np.max([col_coor, 0]), np.min([col_coor + col_val,
+                                                     detector_size[0]])
+        top, bottom = np.max([row_coor, 0]), np.min([row_coor + row_val,
+                                                     detector_size[1]])
+        assert_almost_equal(left, ind_co[0][0])
+        assert_almost_equal(right-1, ind_co[-1][0])
+        assert_almost_equal(top, ind_co[0][1])
+        assert_almost_equal(bottom-1, ind_co[-1][-1])
+
+    assert_array_equal(num_pixels, num_pixels_m)
+
+
+def test_roi_rings():
+    calibrated_center = (6., 4.)
+    img_dim = (20, 25)
+    first_q = 2
+    delta_q = 3
+    num_qs = 7  # number of Q rings
+
+    (q_inds, q_ring_val, num_pixels,
+     pixel_list) = diff_roi.roi_rings(img_dim, calibrated_center, num_qs,
+                                      first_q, delta_q)
+
+    # check the rings edge values
+    q_ring_val_m = np.array([[2, 5], [5, 8], [8, 11], [11, 14], [14, 17],
+                             [17, 20], [20, 23]])
+
+    assert_array_almost_equal(q_ring_val_m, q_ring_val)
+
+    # check the pixel_list and q_inds and num_pixels
+    _helper_check(pixel_list, q_inds, num_pixels, q_ring_val,
+                  calibrated_center, img_dim, num_qs)
```

Comment:
Fixed the problem with the tests
![roi_rings](https://cloud.githubusercontent.com/assets/5701652/6286257/5e4ba304-b8d3-11e4-812b-4e77e988dc72.png)